src/components/MessageForm/index.tsx

```diff
@@ -1,21 +1,43 @@
+import { useRef } from 'react';
+import { ChatType } from '../../type/chat';
+
+type CallbackType = (prev: ChatType[]) => ChatType[];
+
 type Props = {
-  socket: any
+  socket: any,
+  setChats: (prev: CallbackType) => void,
+  name: string
 };
 
-export default function MessageForm({ socket }: Props) {
+export default function MessageForm({ socket, setChats, name }: Props) {
+  const input = useRef<HTMLInputElement>(null);
+
   const submitHandler = (e: React.FormEvent) => {
     e.preventDefault();
 
     const formData = new FormData(e.currentTarget as HTMLFormElement);
 
     (Array.from(formData.values()) as string[]).forEach((message) => {
       socket.emit('chat', message);
+
+      setChats((prev: ChatType[]) => ([
+        ...prev,
+        {
+          id: socket.id,
+          name,
+          message,
+        },
+      ]));
     });
+
+    if (input.current) {
+      input.current.value = '';
+    }
```

Comment:
input.current 가 없는 경우도 있나요?

Reply:
네 null 체킹 안하면 타입스크립트에서 빨간줄로 오류떠요

Reply:
그럼 없는 경우가 아니에요 ㅎㅎ

```suggestion
    input.current!.value = '';
```

!는 있으니까 그냥 써라 라는거에요